platform/src/main/java/org/hillview/storage/CassandraSSTableLoader.java

```diff
@@ -171,137 +230,269 @@ public Schema getSchema(CFMetaData metadata) {
+                                            break;
+                                        case DATE:
+                                            long msTime = SimpleDateSerializer
+                                                    .dayToTimeInMillis(ByteBufferUtil.toInt(byteBuff));
+                                            col.append(Instant.ofEpochMilli(msTime));
+                                            break;
+                                        case DURATION:
+                                            Duration duration = DurationSerializer.instance.deserialize(byteBuff);
+                                            if (duration.getMonths() == 0) {
+                                                int days = duration.getDays();
+                                                long nanos = duration.getNanoseconds();
+                                                long millis = (nanos + Duration.NANOS_PER_HOUR * 24 * days) / 1000000;
+                                                col.append(Converters.toDuration(millis));
+                                            } else {
+                                                // java.time.Duration support day and time (but not month and year)
+                                                throw new RuntimeException(
+                                                        "Failed to convert Duration with months != 0");
+                                            }
+                                            break;
+                                        case EMPTY:
+                                        default:
+                                            throw new RuntimeException(
+                                                    "Unhandled column type " + colType.toString());
+                                    }
+                                }
+                                currentColumn++;
+                            }
+                            // Missing column case #3
+                            // Where unloaded missing column(s) is AFTER last non missing column
+                            while (currentColumn != colCount) {
+                                col = columns.get(currentColumn);
+                                col.appendMissing();
+                                currentColumn++;
                             }
-                            col = columns.get(currentColumn);
-                            col.append(value);
-                            currentColumn++;
                         }
-                        i++;
-                    }
-                } // else tombstone marker
+                    });
+                ;
             }
-        });
-        return columns;
+        }
+    }
+
+    private static boolean bigDecimalInRange(BigDecimal bigDecimal) {
+        org.apache.commons.lang3.Range<BigDecimal> doubleRange = org.apache.commons.lang3.Range
```

Comment:
This is extremely expensive: you are allocating an object for every row.
Also, make this function return the double rather than a boolean and throw otherwise.